src/core/config/config.go

```diff
@@ -149,6 +152,17 @@ func RegisterFlags() {
 		if err := Viper.BindPFlag(strings.ReplaceAll(flag.Name, "-", "_"), fs.Lookup(flag.Name)); err != nil {
 			return
 		}
+
+		oldKey := strings.ToUpper(OldEnvPrefix + "_" + strings.ReplaceAll(flag.Name, "-", "_"))
+		newKey := strings.ToUpper(NewEnvPrefix + "_" + strings.ReplaceAll(flag.Name, "-", "_"))
```

Comment:
Generally it's good practice to use `fmt.Sprintf()` for building strings instead of "string" + "another_string". However, I think your solution is more readable so it's fine here.

I would perhaps reuse the`agent_config.KeyDelimiter` variable we have here:
```suggestion
		oldKey := strings.ToUpper(
			OldEnvPrefix + agent_config.KeyDelimiter + strings.ReplaceAll(flag.Name, "-", agent_config.KeyDelimiter)
		)
		newKey := strings.ToUpper(
			NewEnvPrefix + agent_config.KeyDelimiter + strings.ReplaceAll(flag.Name, "-", agent_config.KeyDelimiter)
		)
```